### ISSUE TITLE
Recommendation on inference (tuning !?)

### ISSUE THREAD

**Issue by thtranos** [lchdl/annotation_free_wmh_seg#4]
Greetings! 

I just upgraded my pc successfully and now I am able to run consistently experiments with your tool and that's really awesome!

I would like to make some questions:

1. What's the difference of raw - 3mm or fov? Which one should I use that refers to the actual image out of the folders located at /DeepWMH/output/002_Segmentations/?
2. Is there any hyperparameter that I could adjust during inference? I think I've read somewhere in the paper that for example a threshold value is automatically adjustable, is there anything that I could configure and test?
3. Can you train further a pretrained model of yours based on a new dataset? And if so, would you suggest it?
4. From what I understand, from the paper again, is that I should input FLAIR of whatever slice length?
5. What tool do you use to open the gif generated from the preview folder (output) it seems that it will not open for me as it supposedly contains errors.
6. The segmentation mask produced is binary (black and white)? Or is there any configuration that allows different colours (like presented - dark yellow, light yellow, purple, red, dark red etc.) which will represent the low/high chance of anomaly?

I hope I am not bothering you too much! Please take your time and answer whenever and on whatever that it is possible!

Theodore Tranos

**Comment by lchdl** [comment]
Hi @thtranos, 
I am really happy to hear about your positive experience using our tool! If you like, you can give a star to this repository :).
1. The folders under `002_Segmentations/` represent the WMH segmentation after each post-processing stage:
    * The **raw** segmentation is the output directly produced from the network, without any post-processing applied.
    * The **3mm** segmentation result is obtained by removing all lesion components with a volume less than 3mm^3.
    * The **fov** segmentation result is obtained by further constraining the lesions to be within the brain mask (the brain mask is produced by ROBEX); any lesion predictions outside the brain mask will be ignored.
2. The entire pipeline runs without any hyperparameter tuning (p.s.: that is also a selling point of our method) in both training and inference stages. The threshold value is automatically determined for each patient image during the training stage, and the adjustable threshold you mentioned is just an ablation study to test if the automatic thresholding strategy is robust enough.
3. I suggest doing so. If the model is trained with images acquired from more data centers and MR scanners, it will become more robust. However, currently, I don't have enough time to do this :(.
    * Actually, you can train your own model from scratch using images acquired from multiple data centers. You can also download FLAIR images with WMH lesions from public datasets. Since the pipeline training does not require any manual annotations, all you need to provide are the patient and normal FLAIR images. That's all. I have provided an example of training a network using a publicly available dataset OASIS3. You can look at the source code `DeepWMH/experiments/010_OASIS3/run_Siemens_Biograph_mMR.py` and `DeepWMH/experiments/010_OASIS3/run_Siemens_TrioTim.py` to see what actually happens during training. You can run the entire experiment with just one push of a button (just type `python xxx.py` and wait for it to finish). These scripts also provide an example of training the model using your own data from scratch.
4. Yes, indeed. The pretrained model was trained using FLAIR images with multiple slice thicknesses ranging from 1mm to 6mm. Therefore, the model should be sufficiently robust to handle FLAIR images with different physical resolutions, which is particularly useful for clinical scenarios. However, for optimal performance, I would recommend using FLAIR images with a slice thickness of 1mm.
5. I use **IrfanView** (`https://www.irfanview.com/`) to view GIF images since it can open a wide range of image formats.
6. If you view the segmentation result in MIPAV, you can change the colormap from grayscale (default) to whatever colormap you like. If you decide to process and visualize them using Python, I would suggest using nibabel for NIFTI image I/O. It's lightweight and easy to use. Then, you can visualize them using matplotlib. The color shading part is up to you.
    * Also, note that the segmentation result is binary (0/1), and it represents whether the voxel at the corresponding location belongs to a lesion or not. It is not an anomaly score, which is in a continuous range. Using a colormap on a binary mask may not be as useful compared to images with continuous values.
